core/src/main/scala/spinoco/fs2/cassandra/system/system.scala

```diff
@@ -41,41 +45,37 @@ package object system {
 
   /** checks whether these two columns are of the same name and type **/
   def sameColumnDef(nameA:String, tpeA:DataType)(nameB:String, tpeB:DataType):Boolean = {
-    lazy val argsA = tpeA.getTypeArguments.asScala.map(_.getName)
-    lazy val argsB = tpeB.getTypeArguments.asScala.map(_.getName)
-    lazy val sameArgs:Boolean = {
-      if (argsA.size != argsB.size) false
-      else {
-        (argsA zip argsB).forall{ case (nA,nB) => nA.isCompatibleWith(nB) }
-      }
-    }
+    val typesEqual = tpeA.asCql(true, false) == tpeB.asCql(true, false)
+    val namesEqual = nameA.equalsIgnoreCase(nameB)
+    namesEqual && typesEqual
+}
 
-    nameA.equalsIgnoreCase(nameB) &&
-      tpeA.getName.isCompatibleWith(tpeB.getName) &&
-      sameArgs
+  /** checks whether the primary keys of given tables are the same **/
+  def samePrimaryKey(current: TableMetadata, desired: AbstractTable[_,_,_,_]):Boolean = {
+    val currentPk = current.getPartitionKey.asScala.map(_.getName.asInternal.toLowerCase).toSeq
+    val desiredPk = desired.partitionKey.map(_.toLowerCase)
+    val currentCk = current.getClusteringColumns.asScala.keys.map(_.getName.asInternal.toLowerCase).toSeq
+    val desiredCk = desired.clusterKey.map(_.toLowerCase)
+     desiredCk == currentCk && desiredPk == currentPk
   }
 
-  /** checks whether the primary keys of given tables are the same **/
-  def samePrimaryKey(current: AbstractTableMetadata, desired: AbstractTable[_,_,_,_]):Boolean = {
-    val currentPk = current.getPartitionKey.asScala.map(_.getName.toLowerCase)
-    val currentCk = current.getClusteringColumns.asScala.map(_.getName.toLowerCase)
+  def samePrimaryKey2(currentPartitioning: List[ColumnMetadata], currentClustering: Map[ColumnMetadata, ClusteringOrder], desired: AbstractTable[_,_,_,_]):Boolean = {
```

Comment:
we need to discuss this "2" variant

Reply:
Since it was there only for the materialized view code, I have removed it...